rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1578,19 +1578,32 @@ public void addSignature(BtcECKey federatorBtcPublicKey, List<byte[]> signatures
 
         Context.propagate(btcContext);
 
+        if (svpIsOngoing()) {
+            provider.getSvpSpendTxWaitingForSignatures()
+                .filter(svpSpendTxWFS -> svpSpendTxWFS.getKey().equals(rskTxHash))
+                .ifPresent(svpSpendTxWFS -> {
+                    logger.trace("Going to add federator public key {} to svp spend transaction", federatorBtcPublicKey);
+                    BtcTransaction svpSpendTx = svpSpendTxWFS.getValue();
+                    if (!areSignaturesEnoughToSignAllTxInputs(svpSpendTx, signatures)) {
+                        return;
+                    }
+                    addSvpSpendTxSignatures(federatorBtcPublicKey, signatures, rskTxHash, svpSpendTx);
+                });
+        }
+
         BtcTransaction releaseTx = provider.getPegoutsWaitingForSignatures().get(rskTxHash);
         if (releaseTx == null) {
             logger.warn("No tx waiting for signature for hash {}. Probably fully signed already.", rskTxHash);
             return;
         }
-        if (!hasEnoughSignatures(releaseTx, signatures)) {
+        if (!areSignaturesEnoughToSignAllTxInputs(releaseTx, signatures)) {
             return;
         }
 
         addReleaseSignatures(federatorBtcPublicKey, signatures, rskTxHash, releaseTx);
```

Comment:
I think this whole block can be moved into its own method. Wdyt?

Reply:
The idea is the method name expresses by itself what you doing in those lines of code and your functional code looks more elegant. See [this](https://github.com/rsksmart/powpeg-node/blob/95c32ece6df43039038fa1f22691affdd5f630ee/src/main/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessage.java#L130)